src/python/grpcio/grpc/experimental/aio/_interceptor.py

```diff
@@ -176,14 +166,25 @@ def cancelled(self) -> bool:
         if not self._interceptors_task.done():
             return False
 
-        call = self._interceptors_task.result()
-        return call.cancelled()
+        try:
+            call = self._interceptors_task.result()
```

Comment:
If `result()` is available for an `asyncio.Task`, the value of `done()` should be `True` (see [doc](https://docs.python.org/3/library/asyncio-task.html#asyncio.Task.done)). Hence, I don't think this line can be reached when the Task is in error state. So, the cancellation result returned might be wrong.

On the other hand, if the Task is not finished yet, this call will throw an `InvalidStateError`.

We might need to think this over. Same argument apply to other try-catch sections in this file.

Reply:
> On the other hand, if the Task is not finished yet, this call will throw an InvalidStateError

This is way first what is being done is checking if the task was finished or not by doing this

```python
 if not self._interceptors_task.done():
    return False
```

Only if the task as been finished the `result()` method is called. In case of no exceptions the returned value should be a `UnaryUnaryCall` and this is the reason of having the 

```python
try:
    call = self._interceptors_task.result()
else:
     return call.cancelled()
```

If there was an error, two exceptions are expected the `AioRpcError` or the `asyncio.CancelledError` and for each exception use case we implement the right logic, 
for example in the use case of the `canceled()` method

```python
except AioRpcError:
    return False
except asyncio.CancelledError:
    return True
```

For methods that are coroutines, like the `code` one it's a bit easier since we do not need to check if the task is done and we have to basically _await_ till the task has finished, no matter if with an exception or with an error, for example:


```python
async def initial_metadata(self) -> Optional[MetadataType]:
    try:
        call = await self._interceptors_task
    except AioRpcError as err:
        return err.initial_metadata()
    except asyncio.CancelledError:
        return None
    else:
        return await call.initial_metadata()
```

Theoretically, all use cases should be in the test suite .... so If I'm not missing something the pattern used should work for all of the methods and scenarios.